src/index.tsx

```diff
@@ -1,38 +1,96 @@
-import React, { useState, useEffect } from 'react';
-import ReactDOM from 'react-dom';
-import { Vault } from '@hyperion-framework/vault';
-import { Manifest, ManifestNormalized } from '@hyperion-framework/types';
-import Viewer from './components/Viewer/Viewer';
+import React, { useState, useEffect } from "react";
+import ReactDOM from "react-dom";
+import { ManifestNormalized } from "@hyperion-framework/types";
+import {
+  ViewerProvider,
+  useViewerState,
+  useViewerDispatch,
+} from "context/viewer-context";
+import Viewer from "./components/Viewer/Viewer";
 
 interface Props {
-  id: string;
+  manifestId: string;
 }
+
 const sampleManifest: string =
-  'https://raw.githubusercontent.com/mathewjordan/mirador-playground/main/assets/iiif/manifest/new_airliner_invalid.json';
+  "https://raw.githubusercontent.com/mathewjordan/mirador-playground/main/assets/iiif/manifest/new_airliner.json";
+
+const App: React.FC<Props> = ({ manifestId }) => {
+  return (
+    <ViewerProvider manifestId={manifestId}>
+      <RenderViewer />
+    </ViewerProvider>
+  );
+};
+
+const RenderViewer: React.FC = () => {
+  const dispatch: any = useViewerDispatch();
 
-const App: React.FC<Props> = ({ id }) => {
-  const [manifest, setManifest] = useState<ManifestNormalized | undefined>();
-  const [loaded, setLoaded] = useState(false);
+  /**
+   * Retrieve state set by the wrapping <ViewerProvider/> and make
+   * the normalized manifest available from @hyperion-framework/vault.
+   */
+  const state = useViewerState();
+  const { manifestId, vault, isLoaded } = state;
+  const [loaded] = useState(false);
+  const manifest: ManifestNormalized = vault.fromRef({
+    id: manifestId,
+    type: "Manifest",
+  });
+
+  /**
+   * Loaded manifest and site using @hyperion-framework/vault.
+   */
   useEffect(() => {
-    const vault = new Vault();
     vault
-      .loadManifest(id)
+      .loadManifest(manifestId)
       .then((data) => {
-        setManifest(data);
-        setLoaded(true);
+        dispatch({
+          type: "updateActiveCanvas",
+          canvasId: data.items[0].id,
+        });
       })
       .catch((error) => {
         console.error(`Manifest failed to load: ${error}`);
+      })
+      .finally(() => {
+        dispatch({
+          type: "updateIsLoaded",
+          isLoaded: true,
+        });
       });
   }, [loaded]);
```

Comment:
I think this could work just as well with an empty [ ] (instead of cueing from `loaded` change in value).  An empty array argument to `useEffect()` means it will only run once after the component has rendered.